app/src/main/java/es/wolfi/app/passman/PassmanReceiver.java

```diff
@@ -0,0 +1,64 @@
+package es.wolfi.app.passman;
+
+import android.content.BroadcastReceiver;
+import android.content.ClipData;
+import android.content.ClipboardManager;
+import android.content.Context;
+import android.content.Intent;
+import android.widget.Toast;
+
+import androidx.core.app.NotificationManagerCompat;
+
+import es.wolfi.passman.API.Credential;
+import es.wolfi.passman.API.Vault;
+
+public class PassmanReceiver extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        // TODO: This method is called when the BroadcastReceiver is receiving
+        // an Intent broadcast.
+        String intentAction = intent.getAction();
+
+        if (intentAction != null) {
+            if (intentAction.startsWith("COPY")) {
+                String credGuid = intent.getStringExtra("CredGuid");
+                String vaultGuid = intent.getStringExtra("VaultGuid");
+
+                if (credGuid != null && vaultGuid != null) {
+                    Vault v = Vault.getVaultByGuid(vaultGuid);
+                    if (v != null) {
+                        Credential c = v.findCredentialByGUID(credGuid);
+                        if (c != null) {
+                            switch (intentAction) {
+                                case "COPYUSERNAMEINTENTACTION":
```

Comment:
It would be nice to have the intent actions as class constants. It makes them easier to remember and reference from outside of this scope.

Reply:
fixed in ac69ecafae0616bc14fae1e4fb027e515719dcc9